src/chart/generateCategoricalChart.tsx

```diff
@@ -1771,58 +1767,23 @@ export const generateCategoricalChart = ({
         this.state;
       const tooltipEventType = this.getTooltipEventType();
       // The cursor is a part of the Tooltip, and it should be shown (by default) when the Tooltip is active.
-      const isActive = element.props.active ?? isTooltipActive;
-
-      if (
-        !element ||
-        !element.props.cursor ||
-        !isActive ||
-        !activeCoordinate ||
-        (chartName !== 'ScatterChart' && tooltipEventType !== 'axis')
-      ) {
-        return null;
-      }
+      const isActive: boolean = element.props.active ?? isTooltipActive;
       const { layout } = this.props;
-      let restProps;
-      let cursorComp: React.ComponentType<any> = Curve;
-
-      if (chartName === 'ScatterChart') {
-        restProps = activeCoordinate;
-        cursorComp = Cross;
-      } else if (chartName === 'BarChart') {
-        restProps = getCursorRectangle(layout, activeCoordinate, offset, tooltipAxisBandSize);
-        cursorComp = Rectangle;
-      } else if (layout === 'radial') {
-        const { cx, cy, radius, startAngle, endAngle } = getRadialCursorPoints(activeCoordinate);
-        restProps = {
-          cx,
-          cy,
-          startAngle,
-          endAngle,
-          innerRadius: radius,
-          outerRadius: radius,
-        };
-        cursorComp = Sector;
-      } else {
-        restProps = { points: getCursorPoints(layout, activeCoordinate, offset) };
-        cursorComp = Curve;
-      }
-      const key = element.key || '_recharts-cursor';
-      const cursorProps = {
-        stroke: '#ccc',
-        pointerEvents: 'none',
-        ...offset,
-        ...restProps,
-        ...filterProps(element.props.cursor),
-        payload: activePayload,
-        payloadIndex: activeTooltipIndex,
-        key,
-        className: 'recharts-tooltip-cursor',
-      };
 
-      return isValidElement(element.props.cursor)
-        ? cloneElement(element.props.cursor, cursorProps)
-        : createElement(cursorComp, cursorProps);
+      return (
+        <Cursor
+          activeCoordinate={activeCoordinate}
```

Comment:
@PavelVanecek there is a `key` error here. Run our tests and see it in every test with a Tooltip coming from `renderCursor`. renderByOrder adds these elements to an array and so this needs a key rather than the in `Cursor`